downloader/src/main/kotlin/VersionControlSystem.kt

```diff
@@ -261,10 +261,24 @@ abstract class VersionControlSystem {
                 false
             }
 
-        if (!addGuessedRevision(pkg.id.name, pkg.id.version) && pkg.id.type == "NPM" && pkg.id.namespace.isNotEmpty()) {
-            // Fallback for Lerna workspaces when scoped packages combined with independent versioning are used, e.g.
-            // support Git tag of the format "@organisation/my-component@x.x.x".
-            addGuessedRevision("${pkg.id.namespace}/${pkg.id.name}", pkg.id.version)
+        if (!addGuessedRevision(pkg.id.name, pkg.id.version)) {
+            when {
+                pkg.id.type == "NPM" && pkg.id.namespace.isNotEmpty() -> {
+                    // Fallback for Lerna workspaces when scoped packages combined with independent versioning are used,
+                    // e.g. support Git tag of the format "@organisation/my-component@x.x.x".
+                    addGuessedRevision("${pkg.id.namespace}/${pkg.id.name}", pkg.id.version)
+                }
+
+                pkg.id.type == "GoMod" && pkg.vcsProcessed.path.isNotBlank() -> {
+                    // Fallback for GoMod packages from mono repos which use the tag format described in
+                    // https://golang.org/ref/mod#vcs-version.
+                    val tag = "${pkg.vcsProcessed.path}/${pkg.id.version}"
+
+                    if (workingTree.listRemoteTags().contains(tag)) {
```

Comment:
More Kotlin-ish: `tag in workingTree.listRemoteTags()`.

Also, this could be a one-liner.